pkg/controller/config/process/excluder_test.go

```diff
@@ -0,0 +1,70 @@
+package process
+
+import (
+	"testing"
+)
+
+func TestExactOrPrefixMatch(t *testing.T) {
+	tcs := []struct {
+		name     string
+		nsMap    map[string]bool
+		ns       string
+		excluded bool
+	}{
+		{
+			name: "exact text match",
+			nsMap: map[string]bool{
+				"kube-system": true,
+				"foobar":      false,
+			},
+			ns:       "kube-system",
+			excluded: true,
+		},
+		{
+			name: "exact text matches false",
+			nsMap: map[string]bool{
+				"kube-system": true,
+				"foobar":      false,
+			},
+			ns:       "foobar",
+			excluded: false,
+		},
+		{
+			name: "wildcard prefix match",
+			nsMap: map[string]bool{
+				"kube-*": true,
+				"foobar": false,
+			},
+			ns:       "kube-system",
+			excluded: true,
+		},
+		{
+			name: "wildcard prefix matches false",
+			nsMap: map[string]bool{
+				"gatekeeper-*": true,
+				"kube-*":       false,
+				"foobar":       false,
+			},
+			ns:       "kube-system",
+			excluded: false,
+		},
+		{
+			name: "wildcard prefix mis-matches false",
+			nsMap: map[string]bool{
+				"gatekeeper-*": true,
+				"foobar":       false,
+			},
+			ns:       "kube-system",
+			excluded: false,
+		},
```

Comment:
should we add a wildcard only and empty test here too?

Reply:
Good catch.  I think the best way to deal with this is to implement a pattern requirement on the excludedNamespace field.  I believe I can do that with a kubebuilder annotation.  Will research now.

Reply:
I've added a pattern check on `excludedNamespaces` in the `config` resource's `match` section.  This will _prevent_ both `"*"` and `""` from being added under `excludedNamespaces`.

@sozercan @maxsmythe is that a reasonable way to solve this problem?

Reply:
Does this change count as backwards incompatible?  At least at the level of the CRD, those were valid entries before.  They weren't valid namespaces, so perhaps it's alright...

Reply:
IMO it's fine. "*" and "" never would have matched an existing namespace (invalid namespace names), so there would be no behavioral change that would impact users.

If, for some reason, these values were in user's constraints, they weren't doing anything and they should probably be notified of that as it was likely a mistake.